src/ports.rs

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use std::sync::Arc;
+
+use anyhow::Context;
+use enarx_config::{Config, File, Protocol};
+use lazy_static::lazy_static;
+use tokio::sync::{RwLock, RwLockWriteGuard};
+
+lazy_static! {
+    static ref PORTS_IN_USE: Arc<RwLock<Vec<u16>>> = Arc::new(RwLock::new(vec![]));
```

Comment:
Shouldn't this be a `HashSet` https://doc.rust-lang.org/std/collections/hash_set/struct.HashSet.html ?